api/src/main/java/org/openmrs/module/insuranceclaims/api/model/InsuranceClaim.java

```diff
@@ -0,0 +1,286 @@
+	}
+
+	public void setClaimedTotal(BigDecimal claimedTotal) {
+		this.claimedTotal = claimedTotal;
+	}
+
+	public BigDecimal getApprovedTotal() {
+		return approvedTotal;
+	}
+
+	public void setApprovedTotal(BigDecimal approvedTotal) {
+		this.approvedTotal = approvedTotal;
+	}
+
+	public Date getDateProcessed() {
+		return dateProcessed == null ? null : (Date) dateProcessed.clone();
+	}
+
+	public void setDateProcessed(Date dateProcessed) {
+		this.dateProcessed = dateProcessed == null ? null : (Date) dateProcessed.clone();
+	}
+
+	public String getExplanation() {
+		return explanation;
+	}
+
+	public void setExplanation(String explanation) {
+		this.explanation = explanation;
+	}
+
+	public String getRejectionReason() {
+		return rejectionReason;
+	}
+
+	public void setRejectionReason(String rejectionReason) {
+		this.rejectionReason = rejectionReason;
+	}
+
+	public String getGuaranteeId() {
+		return guaranteeId;
+	}
+
+	public void setGuaranteeId(String guaranteeId) {
+		this.guaranteeId = guaranteeId;
+	}
+
+	public VisitType getVisitType() {
+		return visitType;
+	}
+
+	public void setVisitType(VisitType visitType) {
+		this.visitType = visitType;
+	}
+
+	public InsuranceClaimStatus getClaimStatus() {
+		return claimStatus;
+	}
+
+	public void setClaimStatus(InsuranceClaimStatus claimStatus) {
+		this.claimStatus = claimStatus;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+
+		InsuranceClaim that = (InsuranceClaim) o;
+
+		return new EqualsBuilder()
+				.appendSuper(super.equals(o))
+				.append(id, that.id)
+				.append(provider, that.provider)
+				.append(patient, that.patient)
+				.append(location, that.location)
+				.append(claimCode, that.claimCode)
+				.append(dateFrom, that.dateFrom)
+				.append(dateTo, that.dateTo)
+				.append(adjustment, that.adjustment)
+				.append(claimedTotal, that.claimedTotal)
+				.append(approvedTotal, that.approvedTotal)
+				.append(dateProcessed, that.dateProcessed)
+				.append(explanation, that.explanation)
+				.append(rejectionReason, that.rejectionReason)
+				.append(guaranteeId, that.guaranteeId)
+				.append(visitType, that.visitType)
+				.append(claimStatus, that.claimStatus)
+				.isEquals();
+	}
+
+	@Override
+	public int hashCode() {
+		return new HashCodeBuilder(InsuranceClaimsConstants.HASH_CODE_INITIAL_NON_ZERO_ODD_NUMBER,
+				InsuranceClaimsConstants.HASH_CODE_MULTIPLIER_NON_ZERO_ODD_NUMBER)
+				.appendSuper(super.hashCode())
+				.append(id)
+				.append(provider)
+				.append(patient)
+				.append(location)
+				.append(claimCode)
+				.append(dateFrom)
+				.append(dateTo)
+				.append(adjustment)
+				.append(claimedTotal)
+				.append(approvedTotal)
+				.append(dateProcessed)
+				.append(explanation)
+				.append(rejectionReason)
+				.append(guaranteeId)
+				.append(visitType)
+				.append(claimStatus)
+				.toHashCode();
```

Comment:
org.apache.commons.lang3.builder.HashCodeBuilder ,

e.g. - HashCodeBuilder.reflectionHashCode(this, "id");

the same with EqualsBuilder and move it to AbstractBaseOpenMRSData

Reply:
Great idea, thanks.